java/test/jmri/util/JUnitUtil.java

```diff
@@ -1272,15 +1272,13 @@ public static void resetProfileManager() {
 
     /**
      * Use if the profile needs to be written to or cleared as part of the test.
-     * Suggested use in the {@link org.junit.Before} annotated method is:      <code>
-     *
-     * @Rule
-     * public org.junit.rules.TemporaryFolder folder = new org.junit.rules.TemporaryFolder();
+     * Suggested use in the {@link org.junit.jupiter.api.BeforeEach} annotated method is:      <code>
      *
      * @BeforeEach
-     * public void setUp() {
-     *     resetProfileManager(new jmri.profile.NullProfile(folder.newFolder(jmri.profile.Profile.PROFILE)));
-     * }
+     * public void setUp(@org.junit.jupiter.api.io.TempDir File folder) throws java.io.IOException {
```

Comment:
I don't think this will work. I believe that `@BeforeEach` expects the method it annotates to have no arguments.

I suggest you test this in a real test before changing the Javadoc in JUnitUtil.

Reply:
`@BeforeEach` can definitely have a `@TempDir` field injection , works fine in real tests.
The scope of the injected Temp Directory with `@BeforeEach` tags is discussed at https://junit.org/junit5/docs/5.4.0/api/org/junit/jupiter/api/io/TempDir.html

The `@BeforeEach` tags ( and `@AfterEach`, `@BeforeAll` etc. ) support injection, https://junit.org/junit5/docs/5.4.0/api/org/junit/jupiter/api/BeforeEach.html